lib/workers/testbot.ts

```diff
@@ -177,13 +176,16 @@ export class TestBot {
 	/**
 	 * Flash SD card with operating system
 	 */
-	public async flash(url: string): Promise<void> {
+	public async flash(stream: Stream.Readable): Promise<void> {
 		await this.off();
 
 		await this.criticalSection(async () => {
+			const source = await new sdk.sourceDestination.ZipSource(
```

Comment:
You can use `StreamZipSource` directly here. `ZipSource` is a wrapper that first finds out if the source is randomly accessible then uses `StreamZipSource` or `RandomAccessZipSource`.